pkg/config/dotenv.go

```diff
@@ -0,0 +1,65 @@
+package config
+
+import (
+	"os"
+	"strings"
+
+	log "github.com/golang/glog"
+
+	"github.com/joho/godotenv"
+)
+
+const (
+	errNoSuchFileDirectory = "no such file or directory"
+)
+
+func isNoSuchFileErr(err error) bool {
+	return strings.Contains(err.Error(), errNoSuchFileDirectory)
+}
+
+// PopulateFromDotEnv attempts to retrieve env vars from .env files to populate
+// into the environment.
+// envEnvVar passed in indicates the env var that specifies the environment to
+// use "test", "development", "production" for the .env file
+func PopulateFromDotEnv(envEnvVar string) error {
+	env := os.Getenv(envEnvVar)
+	if "" == env {
+		env = "development"
+	}
+
+	err := godotenv.Load(".env." + env + ".local")
+	if err != nil && !isNoSuchFileErr(err) {
+		log.Errorf("Did not load .env.%v.local: err: %v", env, err)
+		return err
```

Comment:
are you sure returning here is the best behavior? The .local is generally used to override and is not required

Reply:
(same issue on line 42 and 51) wondering if maybe this should be a debug instead of an error

Reply:
So, this particular error return is only in the event that it is *not* a "file not found" error, meaning that the file was loaded, but the data was malformed or some other internal error.  I experienced a situation where I was using a `.env` and while the file was found, the data in the `.env` was malformed. 

Since there was no logging in the old code for the error, just the message that the `.env` file was not loaded, so it was confusing since the file was there.  

So basically in this update, if the file is found but malformed or other, return an error.  If the file is not found, keep going.  If the file is found and is not malformed, keep going and log that the file was loaded.

Does that make sense?

Reply:
oh yes makes complete sense, I was just not looking at the code very carefully. :+1: